### DISCUSSION TITLE
How to recover funds sent to an address on the wrong network

### DISCUSSION THREAD

**Discussion by jpalvarezl** [gnosis/safe#425] [Show and tell]
L2 chains (or side chains or child chains) introduce advantages to users such as lower gas costs and faster blocks. To communicate between between L2 chains with main chains (such as Ethereum Mainnet), "bridges" were introduced. 

Bridges are sets of contracts that allow moving assets from a root chain to a child chain. Different bridges could have different implementations of this behaviour. In the case of the Matic chain, for example, it seems that when a user transfers assets from Ethereum to the Matic chain, no transaction is triggered in the new chain, and the balance in the Matic chain is then topped up at the recipient account in the new chain. The recipient address in the Matic chain is the same as the one in the main chain. (see `References`)

Which introduces the pitfal: a user may top up an address that in the main chain corresponds to their safe; in the L2 chain there might not be a safe deployed to this public address. Rendering the funds inaccessible.

There are tools that a user can use for deploying a safe in the L2 under the address and therefore regain access to the funds. As of the writing of this however, there is no planned support for this in the UI nor an article specifying the migration path of a user for their safe.

The tools that can be leveraged for this purpose are linked in the `Reference` section. The `safe-contract-deployment-replay` to deploy the safe itself, and then in order to interact with it: `safe-tasks` or `safe-cli` (depending on the user preference for language. Not sure what is the state in terms of feature parity between those 2)

### References

MATIC <> Ethereum Bridge: https://docs.matic.network/docs/develop/ethereum-matic/plasma/getting-started/
`safe-contract-deployment-relay`: https://github.com/gnosis/safe-contract-deployment-replay
`safe-cli`: https://github.com/gnosis/safe-cli
`safe-tasks`: https://github.com/gnosis/safe-tasks

**Comment by tschubotz** [comment]
Wiki article on how to obtain funds for sidechains/L2: https://github.com/gnosis/safe/wiki/How-to-get-funds-for-L2-sidechains

**Comment by jpalvarezl** [comment]
Another article of interest: https://www.xdaichain.com/for-users/bridges/omnibridge/omnibridge-+-gnosis-safe-app
